camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -280,20 +285,10 @@ private void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSour
 
     List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs);
 
-    for (Properties jobProps : jobPropsList) {
+    for (Properties jobProps : jobPropsList){
       tasksToComplete.add(runCollector(jobProps, topic));
     }
 
-    for (Future<?> task : tasksToComplete) {
-      try {
-        task.get();
-      } catch (InterruptedException e) {
-        throw e;
-      } catch (ExecutionException e) {
-        throw e;
-      }
-    }
-
```

Comment:
don't we need to wait for the futures to complete anywhere?

> [executorService.shutdown] does not wait for previously submitted tasks to complete execution. Use awaitTermination to do that.

and we wait only for 30secs (see other comment).